projects/core/src/main/kotlin/core/Boat.kt

```diff
@@ -22,17 +37,78 @@ class Boat(private val broadcast: Broadcast, private val propulsionSystem: Propu
     private val killSwitch: Subject<Void, Void> = PublishSubject.create()
 
     fun start(io: Scheduler, clock: Scheduler) {
-        val speeds = broadcast.valuesOfType<Motion>()
-            .startWith(Motion(0.0, 0.0))
-            .map { speed(it) }
-            .onBackpressureLatest()
         val ticks = Observable.interval(SLEEP_DURATION_MS, TimeUnit.MILLISECONDS, clock)
         val positions = Observable.combineLatest(
             broadcast.valuesOfType<GpsFix>(),
             broadcast.valuesOfType<GpsNavInfo>(),
             { fix, nav -> Pair(fix, nav)}
         )
 
+        val currentGps = Observable.combineLatest(
+            broadcast.valuesOfType<GpsFix>(),
+            broadcast.valuesOfType<GpsNavInfo>(),
+            { fix, nav -> Pair(fix, nav)}
```

Comment:
This combiner function could be replaced with `GpsValue.from(nav, fix)` from below